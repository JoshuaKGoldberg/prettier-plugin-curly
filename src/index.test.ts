import * as prettier from "prettier";
import { describe, expect, test } from "vitest";

import * as plugin from "./index.js";

function format(code: string, options: prettier.Options) {
	return prettier.format(code, {
		...options,
		plugins: [...(options.plugins ?? []), plugin],
	});
}

describe("Tests", () => {
	test.each([
		[`do a; while (b);`, `do{a}while(b);`],
		[`do { a; } while (b);`, `do { a; } while (b);`],
		[
			`
do
/* a */
b; while (c)
`,
			`
do{
/* a */
b}while(c);
`,
		],
		[
			`
do /* a */
b; while (c)
`,
			`
do{/* a */
b}while(c);
`,
		],
		[
			`
do
//a
b; while (c)
`,
			`
do{
//a
b}while(c);
`,
		],
		[
			`
do //a
b; while (c)
`,
			`
do{//a
b}while(c);
`,
		],
		[`for (; ; ) ;`, `for (; ; ) ;`],
		[`for (; ; ) d;`, `for(;;){d}`],
		[`for (; ; ) { d; }`, `for (; ; ) { d; }`],
		[`for (a; b; c) d;`, `for(a;b;c){d}`],
		[`for (a; b; c) { d; }`, `for (a; b; c) { d; }`],
		[`for (const a in b) c;`, `for(const a in b){c}`],
		[`for (const a in b) { c; }`, `for (const a in b) { c; }`],
		[`for (const a of b) c;`, `for(const a of b){c}`],
		[`for (const a of b) { c; }`, `for (const a of b) { c; }`],
		[`if (a) b;`, `if(a){b}`],
		[`if (a) { b; }`, `if (a) { b; }`],
		[`if (a) b; else c;`, `if(a){b}else{c}`],
		[`if (a) {} else b;`, `if(a){}else{b}`],
		[`if (a) b; else { c; }`, `if(a){b}else{c}`],
		[`if (a) b; else if (b) c;`, `if(a){b}else if(b){c}`],
		["if (a) if (b) c;", "if(a){if(b){c}}"],
		["if (a) { if (b) c; }", "if (a) { if(b){c} }"],
		["if (a) if (b) if (c) d;", "if(a){if(b){if(c){d}}}"],
		["if (a) if (b) {} else if (c) d;", "if(a){if(b){}else if(c){d}}"],
		[
			`
if(a) //b
c
`,
			`
if(a){//b
c}`,
		],
		[
			`
if(a)
//b
c
`,
			`
if(a){
//b
c}`,
		],
		[`let a; let a;`, `let a; let a;`],
		[`foo; import a from 'bar'`, `foo; import a from 'bar'`],
		[`return;`, `return;`],
		[`new.target;`, `new.target;`],
		[`await 1;`, `await 1;`],
		// [`super;`, `super;`],
		[`export { foo };`, `export { foo };`],
		[`while (a) ;`, `while (a) ;`],
		[`while (a) b;`, `while(a){b}`],
		[`while (a) { b; }`, `while (a) { b; }`],
		[`while (a) <b>c;`, `while(a){<b>c}`],
		[`while (a) <b />;`, `while(a){<b/>}`, `test.js`],
		[`while (a) <b />;`, `while(a){<b/>}`, `test.tsx`],
		[
			`
while (a) // b
<c> d;
`,
			`
while(a){// b
<c>d}
`,
		],

		// Babel has different behaviors around newlines which we should avoid.
		// This test ensures irrelevant nodes don't get reformatted unnecessarily.
		// https://github.com/JoshuaKGoldberg/prettier-plugin-curly/pull/309/files#r1527185682
		[
			`
if (a) b;
_ = {
	a: [
		"b",
	],
	c: "d"
};
`,
			`
if(a){b}
_ = {
	a: [
		"b",
	],
	c: "d"
};
`,
		],
		[
			`
const p = 'test';


//pre comment
//this is another pre comment
if (a) b;
//post comment

someOtherCode();
`,
			`
const p = 'test';


//pre comment
//this is another pre comment
if(a){b}
//post comment

someOtherCode();
`,
		],
		[
			`
// pre comment
if (a)
	// inner comment
	b;
// post comment
`,
			`
// pre comment
if(a){
// inner comment
b}
// post comment
`,
		],
		// #625
		[
			`
// prettier-ignore
function ignored() {
    if(foo  .bar)
        return
}
`,
			"To be filled",
		],
	])("case", async (input, expected, filepath = "test.ts") => {
		const output = await format(input, { filepath });
		const snapshot = `
${createLine(" Input ")}
${input.trimEnd()}

${createLine(" Output ")}
${output.trimEnd()}
`;
		expect(snapshot).toMatchSnapshot();
	});
});

const createLine = (text = "") => {
	const space = 80 - text.length;
	const before = Math.floor(space / 2);
	const after = space - before;

	return "-".repeat(before) + text + "-".repeat(after);
};
