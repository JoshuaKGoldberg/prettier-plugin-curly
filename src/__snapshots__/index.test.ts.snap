// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests > case 1`] = `
"
------------------------------------ Input -------------------------------------
do a; while (b);

------------------------------------ Output ------------------------------------
do {
  a;
} while (b);
"
`;

exports[`Tests > case 2`] = `
"
------------------------------------ Input -------------------------------------
do { a; } while (b);

------------------------------------ Output ------------------------------------
do {
  a;
} while (b);
"
`;

exports[`Tests > case 3`] = `
"
------------------------------------ Input -------------------------------------

do
/* a */
b; while (c)

------------------------------------ Output ------------------------------------
do {
  /* a */
  b;
} while (c);
"
`;

exports[`Tests > case 4`] = `
"
------------------------------------ Input -------------------------------------

do /* a */
b; while (c)

------------------------------------ Output ------------------------------------
do {
  /* a */
  b;
} while (c);
"
`;

exports[`Tests > case 5`] = `
"
------------------------------------ Input -------------------------------------

do
//a
b; while (c)

------------------------------------ Output ------------------------------------
do {
  //a
  b;
} while (c);
"
`;

exports[`Tests > case 6`] = `
"
------------------------------------ Input -------------------------------------

do //a
b; while (c)

------------------------------------ Output ------------------------------------
do {
  //a
  b;
} while (c);
"
`;

exports[`Tests > case 7`] = `
"
------------------------------------ Input -------------------------------------
for (; ; ) ;

------------------------------------ Output ------------------------------------
for (;;);
"
`;

exports[`Tests > case 8`] = `
"
------------------------------------ Input -------------------------------------
for (; ; ) d;

------------------------------------ Output ------------------------------------
for (;;) {
  d;
}
"
`;

exports[`Tests > case 9`] = `
"
------------------------------------ Input -------------------------------------
for (; ; ) { d; }

------------------------------------ Output ------------------------------------
for (;;) {
  d;
}
"
`;

exports[`Tests > case 10`] = `
"
------------------------------------ Input -------------------------------------
for (a; b; c) d;

------------------------------------ Output ------------------------------------
for (a; b; c) {
  d;
}
"
`;

exports[`Tests > case 11`] = `
"
------------------------------------ Input -------------------------------------
for (a; b; c) { d; }

------------------------------------ Output ------------------------------------
for (a; b; c) {
  d;
}
"
`;

exports[`Tests > case 12`] = `
"
------------------------------------ Input -------------------------------------
for (const a in b) c;

------------------------------------ Output ------------------------------------
for (const a in b) {
  c;
}
"
`;

exports[`Tests > case 13`] = `
"
------------------------------------ Input -------------------------------------
for (const a in b) { c; }

------------------------------------ Output ------------------------------------
for (const a in b) {
  c;
}
"
`;

exports[`Tests > case 14`] = `
"
------------------------------------ Input -------------------------------------
for (const a of b) c;

------------------------------------ Output ------------------------------------
for (const a of b) {
  c;
}
"
`;

exports[`Tests > case 15`] = `
"
------------------------------------ Input -------------------------------------
for (const a of b) { c; }

------------------------------------ Output ------------------------------------
for (const a of b) {
  c;
}
"
`;

exports[`Tests > case 16`] = `
"
------------------------------------ Input -------------------------------------
if (a) b;

------------------------------------ Output ------------------------------------
if (a) {
  b;
}
"
`;

exports[`Tests > case 17`] = `
"
------------------------------------ Input -------------------------------------
if (a) { b; }

------------------------------------ Output ------------------------------------
if (a) {
  b;
}
"
`;

exports[`Tests > case 18`] = `
"
------------------------------------ Input -------------------------------------
if (a) b; else c;

------------------------------------ Output ------------------------------------
if (a) {
  b;
} else {
  c;
}
"
`;

exports[`Tests > case 19`] = `
"
------------------------------------ Input -------------------------------------
if (a) {} else b;

------------------------------------ Output ------------------------------------
if (a) {
} else {
  b;
}
"
`;

exports[`Tests > case 20`] = `
"
------------------------------------ Input -------------------------------------
if (a) b; else { c; }

------------------------------------ Output ------------------------------------
if (a) {
  b;
} else {
  c;
}
"
`;

exports[`Tests > case 21`] = `
"
------------------------------------ Input -------------------------------------
if (a) b; else if (b) c;

------------------------------------ Output ------------------------------------
if (a) {
  b;
} else if (b) {
  c;
}
"
`;

exports[`Tests > case 22`] = `
"
------------------------------------ Input -------------------------------------
if (a) if (b) c;

------------------------------------ Output ------------------------------------
if (a) {
  if (b) {
    c;
  }
}
"
`;

exports[`Tests > case 23`] = `
"
------------------------------------ Input -------------------------------------
if (a) { if (b) c; }

------------------------------------ Output ------------------------------------
if (a) {
  if (b) {
    c;
  }
}
"
`;

exports[`Tests > case 24`] = `
"
------------------------------------ Input -------------------------------------
if (a) if (b) if (c) d;

------------------------------------ Output ------------------------------------
if (a) {
  if (b) {
    if (c) {
      d;
    }
  }
}
"
`;

exports[`Tests > case 25`] = `
"
------------------------------------ Input -------------------------------------
if (a) if (b) {} else if (c) d;

------------------------------------ Output ------------------------------------
if (a) {
  if (b) {
  } else if (c) {
    d;
  }
}
"
`;

exports[`Tests > case 26`] = `
"
------------------------------------ Input -------------------------------------

if(a) //b
c

------------------------------------ Output ------------------------------------
if (a) {
  //b
  c;
}
"
`;

exports[`Tests > case 27`] = `
"
------------------------------------ Input -------------------------------------

if(a)
//b
c

------------------------------------ Output ------------------------------------
if (a) {
  //b
  c;
}
"
`;

exports[`Tests > case 28`] = `
"
------------------------------------ Input -------------------------------------
let a; let a;

------------------------------------ Output ------------------------------------
let a;
let a;
"
`;

exports[`Tests > case 29`] = `
"
------------------------------------ Input -------------------------------------
foo; import a from 'bar'

------------------------------------ Output ------------------------------------
foo;
import a from "bar";
"
`;

exports[`Tests > case 30`] = `
"
------------------------------------ Input -------------------------------------
return;

------------------------------------ Output ------------------------------------
return;
"
`;

exports[`Tests > case 31`] = `
"
------------------------------------ Input -------------------------------------
new.target;

------------------------------------ Output ------------------------------------
new.target;
"
`;

exports[`Tests > case 32`] = `
"
------------------------------------ Input -------------------------------------
await 1;

------------------------------------ Output ------------------------------------
await 1;
"
`;

exports[`Tests > case 33`] = `
"
------------------------------------ Input -------------------------------------
export { foo };

------------------------------------ Output ------------------------------------
export { foo };
"
`;

exports[`Tests > case 34`] = `
"
------------------------------------ Input -------------------------------------
while (a) ;

------------------------------------ Output ------------------------------------
while (a);
"
`;

exports[`Tests > case 35`] = `
"
------------------------------------ Input -------------------------------------
while (a) b;

------------------------------------ Output ------------------------------------
while (a) {
  b;
}
"
`;

exports[`Tests > case 36`] = `
"
------------------------------------ Input -------------------------------------
while (a) { b; }

------------------------------------ Output ------------------------------------
while (a) {
  b;
}
"
`;

exports[`Tests > case 37`] = `
"
------------------------------------ Input -------------------------------------
while (a) <b>c;

------------------------------------ Output ------------------------------------
while (a) {
  <b>c;
}
"
`;

exports[`Tests > case 38`] = `
"
------------------------------------ Input -------------------------------------
while (a) <b />;

------------------------------------ Output ------------------------------------
while (a) {
  <b />;
}
"
`;

exports[`Tests > case 39`] = `
"
------------------------------------ Input -------------------------------------
while (a) <b />;

------------------------------------ Output ------------------------------------
while (a) {
  <b />;
}
"
`;

exports[`Tests > case 40`] = `
"
------------------------------------ Input -------------------------------------

while (a) // b
<c> d;

------------------------------------ Output ------------------------------------
while (a) {
  // b
  <c>d;
}
"
`;

exports[`Tests > case 41`] = `
"
------------------------------------ Input -------------------------------------

if (a) b;
_ = {
	a: [
		"b",
	],
	c: "d"
};

------------------------------------ Output ------------------------------------
if (a) {
  b;
}
_ = {
  a: ["b"],
  c: "d",
};
"
`;

exports[`Tests > case 42`] = `
"
------------------------------------ Input -------------------------------------

const p = 'test';


//pre comment
//this is another pre comment
if (a) b;
//post comment

someOtherCode();

------------------------------------ Output ------------------------------------
const p = "test";

//pre comment
//this is another pre comment
if (a) {
  b;
}
//post comment

someOtherCode();
"
`;

exports[`Tests > case 43`] = `
"
------------------------------------ Input -------------------------------------

// pre comment
if (a)
	// inner comment
	b;
// post comment

------------------------------------ Output ------------------------------------
// pre comment
if (a) {
  // inner comment
  b;
}
// post comment
"
`;
